### ISSUE TITLE
app memory continues to grow in each release

### ISSUE THREAD

**Issue by talavidan4** [Microsoft/cordova-plugin-code-push#233]
Hi there,
I just start to use the plugin and notich that each release adding significant amount of memory to the app.
it seems like after updating - the old files are still exists in the device memory.
the memory size, although i'm changing one file for the release, is increasing much more than
the change it self.

does someone familiar with the issue?

**Comment by talavidan4** [comment]
strange - after few more releases, the app memory got back approximately to
the original memory size on the first installation.

**Comment by sergey-akhalkov** [comment]
@talavidan4, due to CodePush has [rollback feature](https://github.com/Microsoft/code-push/blob/89cbef2e96477c7cd52369a261ec69ae6e735867/cli/README.md#rolling-back-updates), it is important to save old versions of code, otherwise CodePush SDK would not able to perform rollbacks. It means that each time new update is downloaded by CodePush SDK - [new folder is created to store it](https://github.com/Microsoft/cordova-plugin-code-push/blob/009e349ca61e5a8924a735a92a08c5e688f4e330/bin/www/localPackage.js#L40). That's why app size is rising while downloading and installing updates.

>strange - after few more releases, the app memory got back approximately to
the original memory size on the first installation.

In case you've installed new binary version of the app (perform `cordova build` command) [CodePush SDK removes all installed updates](https://github.com/Microsoft/cordova-plugin-code-push/blob/3fe22769745a5112b163dc857fab58e9037cef9c/src/android/CodePush.java#L287). That's why the app memory got back approximately to the original memory size on the first installation.

Please let me know if it helps or if you have any questions.

**Comment by talavidan4** [comment]
@sergey-akhalkov - thank you for the reply.
I guessed that it related to the previous versions for rolling back the app.
I'm building the app using PGB, how does it effect my development?
i want to build the app once with PGB, and after it - updating the code using this plugin.

**Comment by sergey-akhalkov** [comment]
@talavidan4, could you please clarify what you mean by 
>  I'm building the app using PGB, how does it effect my development?

In a nutshell, [CodePush supports PGB](https://github.com/Microsoft/cordova-plugin-code-push#phonegap-build), so please let us know if you see any issues.

**Comment by talavidan4** [comment]
@sergey-akhalkov - i'm not using the command `cordova build` for 
building my app, i let PGB do it for me.
so after installing the app from PGB, i am pushing update using the command:
`code-push release-cordova MyApp platform` through cordova cli.
is there option that from this process all the rollback version folders at the device will always remain?

**Comment by sergey-akhalkov** [comment]
@talavidan4, rollback version folders will always remain until you perform new apk build (it does not matter if you've used cordova build or PGB). In case new apk build has been installed - all rollback version folders will be removed due to [this lines of code](https://github.com/Microsoft/cordova-plugin-code-push/blob/3fe22769745a5112b163dc857fab58e9037cef9c/src/android/CodePush.java#L287).

**Comment by talavidan4** [comment]
@sergey-akhalkov - so if i'm installing the app one time from PGB, and doing all the update
using code-push without installing a new apk, is this can lead to increasing
memory size on the device on each release?

**Comment by sergey-akhalkov** [comment]
@talavidan4, I'm sorry for the confusion, I've just tried to install several code push updates and I could see, that the app size is between 3 and 4 MB across all installations, but cache size is rising after each update. I've also found [the logic](https://github.com/Microsoft/cordova-plugin-code-push/blob/adb82ddc2dfa1cf03128eb4db8137cce081c3798/src/android/CodePush.java#L137), that removes old packages after each successful CodePush update installation.
It means that in case you are using CodePush to produce updates - the app size will not rise infinitely.
